src/components/views/settings/ChangePassword.tsx

```diff
@@ -306,13 +303,7 @@ export default class ChangePassword extends React.Component<IProps, IState> {
 
         const oldPassword = this.state.oldPassword;
         const newPassword = this.state.newPassword;
-        const confirmPassword = this.state.newPasswordConfirm;
-        const err = this.checkPassword(oldPassword, newPassword, confirmPassword);
```

Comment:
Removed the `checkPassword(...)` call because it's not necessary and provides no extra protection.

Both of the error cases that `checkPassword(...)` looks at are covered by the logic in `verifyFieldsBeforeSubmit()` which we call just above. There is no way `checkPassword` would ever throw one of these errors since they are already valid by the time it reaches here.

Here is proof that the field validation already covers our butts:

<img width="596" alt="Field validation: Passwords can't be empty" src="https://user-images.githubusercontent.com/558581/232179902-1716686a-aea2-49cf-99d3-bf2c64d4cfa9.png">
<img width="591" alt="Field validation: Passwords don't match" src="https://user-images.githubusercontent.com/558581/232179901-f5169bcc-3551-4f7b-9d36-9c67a5f4694d.png">
<img width="570" alt="Field validation when confirm password is empty: Confirm password" src="https://user-images.githubusercontent.com/558581/232179903-440c022e-6494-4d45-89b7-45bcbb99b2dc.png">

Reply:
In order to move this PR forward without having to spend a bunch of time writing some Cypress tests just to sanity check if password changing still works, I've restored this redundant logic ⏩